build/release/binaries.sh

```diff
@@ -18,48 +46,53 @@ build() {
     if [[ ${os} == "linux" ]]; then
         files+=( rookd${ext} )
 
-        local tarfile=${RELEASE_DIR}/rook-${version}-${os}-${arch}.tar.gz
+        local tarfile=$(get_archive_name $os $arch)
         echo creating tar ${tarfile}
-        tar czf "${tarfile}" -C "${bindir}" ${files[*]}
+        tar czf "${RELEASE_DIR}/${tarfile}" -C "${bindir}" ${files[*]}
 
         # create a package with debug symbols
         files=( rookd${ext}.debug )
-        local tarfile=${RELEASE_DIR}/rook-${version}-${os}-${arch}-debug.tar.gz
+        local tarfile=$(get_archive_name $os $arch "debug")
         echo creating debug tar ${tarfile}
-        tar czf "${tarfile}" -C "${bindir}" ${files[*]}
+        tar czf "${RELEASE_DIR}/${tarfile}" -C "${bindir}" ${files[*]}
     else
-        local zipfile=$(realpath ${RELEASE_DIR}/rook-${version}-${os}-${arch}.zip)
+        local zipfile=$(get_archive_name $os $arch)
+        local zippath=$(realpath ${RELEASE_DIR}/${zipfile})
         echo creating zip ${zipfile}
-        $(cd ${bindir} && zip -qr ${zipfile} ${files[*]})
+        $(cd ${bindir} && zip -qr ${zippath} ${files[*]})
     fi
 }
 
 publish() {
     local os=$1
     local arch=$2
-    local version=${RELEASE_VERSION}
+    local file=$(get_archive_name $os $arch)
 
-    local file=${RELEASE_DIR}/rook-${version}-${os}-${arch}
-    local ext=tar.gz
+    echo uploading $file to S3
+    s3_upload ${RELEASE_DIR}/$file
 
-    if [[ ${os} != "linux" ]]; then
-        ext=zip
+    if [[ ${os} == "linux" ]]; then
+        file=$(get_archive_name $os $arch "debug")
+        echo uploading $file to S3
+        s3_upload ${RELEASE_DIR}/$file
     fi
+}
 
-    echo uploading $file.$ext to S3
-    s3_upload $file.$ext
+promote() {
+    local os=$1
+    local arch=$2
+    local file=$(get_archive_name $os $arch)
+    local ext=$(get_archive_ext $os)
+
+    echo promoting $file to channel ${RELEASE_CHANNEL}
+    s3_promote_file $file
+    github_upload ${RELEASE_DIR}/$file $ext
```

Comment:
help me remember/understand the split between items that get uploaded to s3 and github.  Some appear to only go to s3 but the promote step is also uploading to github.